src/workflow.ts

```diff
@@ -325,10 +335,18 @@ export function getCategoryInput(workflow: Workflow): string | undefined {
       "Could not get category input since multiple categories were specified by the analysis step."
     );
   }
-  if (categories[0].includes("${{")) {
+
+  // Make a basic attempt to substitute matrix variables
+  // First normalize by removing whitespace
+  let category = categories[0].replace(/\${{\s+/, "${{").replace(/\s+}}/, "}}");
```

Comment:
Maybe rare since by default, unless there are explicit quotes, yaml will trim whitespace.

```suggestion
  let category = categories[0].trim().replace(/\${{\s+/, "${{").replace(/\s+}}/, "}}");
```

Reply:
You're right about explicit quotes.  I think if we do have leading / trailing whitespace at this point, then that's the category we'd submit to code scanning in the event of a successful run, so that's what we need to submit for a failed run too.